consensus/processor.go

```diff
@@ -455,6 +447,14 @@ func (p *DefaultProcessor) attemptApplyMessage(ctx context.Context, st *state.Ca
 		}, err
 	}
 
+	// Processing an external message from an empty actor upgrades it to an account actor.
+	if !fromActor.Code.Defined() {
```

Comment:
I should have done this when I originally wrote this code, but it would be great to have an isEmpty() method on actor (or a free function) rather than relying on comments to define the concept of an empty actor.